--- service/src/main/java/cleveres/tricky/cleverestech/Config.kt
+++ service/src/main/java/cleveres/tricky/cleverestech/Config.kt
@@ -619,7 +619,7 @@
                                           value.contains("DD")) {
                             null
                         } else {
-                            runCatching { value.convertPatchLevel(false) }.getOrNull()
+                            runCatching { value.convertPatchLevel(false) }.getOrNull() ?: value
                         }
                         newPatch[key] = preCalc ?: value
                     } else if (parts.size == 1) {
@@ -631,7 +631,7 @@
                                           value.contains("DD")) {
                             null
                         } else {
-                            runCatching { value.convertPatchLevel(false) }.getOrNull()
+                            runCatching { value.convertPatchLevel(false) }.getOrNull() ?: value
                         }
                          newDefault = preCalc ?: value
                     }
