--- a/lsplt/src/main/jni/syscall.hpp
+++ b/lsplt/src/main/jni/syscall.hpp
@@ -4643,18 +4643,30 @@
     #define LSS_SC_LOADARGS_5(arg1, arg2, arg3, arg4, arg5)                   \
         LSS_SC_LOADARGS_4(arg1, arg2, arg3, arg4);                            \
         __sc_8  = (unsigned long) (arg5)
+    #undef LSS_SC_LOADARGS_6
+    #define LSS_SC_LOADARGS_6(arg1, arg2, arg3, arg4, arg5, arg6)             \
+        LSS_SC_LOADARGS_5(arg1, arg2, arg3, arg4, arg5);                      \
+        __sc_9  = (unsigned long) (arg6)
+    #undef LSS_SC_ASMINPUT_0
+    #define LSS_SC_ASMINPUT_0 "0" (__sc_0), "1" (__sc_3)
+    #undef LSS_SC_ASMINPUT_1
+    #define LSS_SC_ASMINPUT_1 LSS_SC_ASMINPUT_0, "2" (__sc_4)
+    #undef LSS_SC_ASMINPUT_2
+    #define LSS_SC_ASMINPUT_2 LSS_SC_ASMINPUT_1, "3" (__sc_5)
+    #undef LSS_SC_ASMINPUT_3
+    #define LSS_SC_ASMINPUT_3 LSS_SC_ASMINPUT_2, "4" (__sc_6)
+    #undef LSS_SC_ASMINPUT_4
+    #define LSS_SC_ASMINPUT_4 LSS_SC_ASMINPUT_3, "5" (__sc_7)
+    #undef LSS_SC_ASMINPUT_5
+    #define LSS_SC_ASMINPUT_5 LSS_SC_ASMINPUT_4, "6" (__sc_8)
+    #undef LSS_SC_ASMINPUT_6
+    #define LSS_SC_ASMINPUT_6 LSS_SC_ASMINPUT_5, "7" (__sc_9)
     #undef LSS_SC_BODY
     #define LSS_SC_BODY(nr, type, opt, args...)                               \
         long __sc_ret, __sc_err;                                              \
         {                                                                     \
           register unsigned long __sc_0 __asm__ ("r0") = __NR_socketcall;     \
           register unsigned long __sc_3 __asm__ ("r3") = opt;                 \
           register unsigned long __sc_4 __asm__ ("r4");                       \
           register unsigned long __sc_5 __asm__ ("r5");                       \
           register unsigned long __sc_6 __asm__ ("r6");                       \
           register unsigned long __sc_7 __asm__ ("r7");                       \
           register unsigned long __sc_8 __asm__ ("r8");                       \
+          register unsigned long __sc_9 __asm__ ("r9");                       \
           LSS_SC_LOADARGS_##nr(args);                                         \
           __asm__ __volatile__                                                \
               ("stwu 1, -48(1)\n\t"                                           \
@@ -4664,15 +4676,16 @@
                "stw 6, 28(1)\n\t"                                             \
                "stw 7, 32(1)\n\t"                                             \
                "stw 8, 36(1)\n\t"                                             \
+               "stw 9, 40(1)\n\t"                                             \
                "addi 4, 1, 20\n\t"                                            \
                "sc\n\t"                                                       \
                "mfcr %0"                                                      \
                  : "=&r" (__sc_0),                                            \
                    "=&r" (__sc_3), "=&r" (__sc_4),                            \
                    "=&r" (__sc_5), "=&r" (__sc_6),                            \
-                   "=&r" (__sc_7), "=&r" (__sc_8)                             \
-                 : LSS_ASMINPUT_##nr                                          \
+                   "=&r" (__sc_7), "=&r" (__sc_8), "=&r" (__sc_9)             \
+                 : LSS_SC_ASMINPUT_##nr                                       \
                  : "cr0", "ctr", "memory");                                   \
           __sc_ret = __sc_3;                                                  \
           __sc_err = __sc_0;                                                  \
@@ -4686,14 +4699,11 @@
                                          int flags) {
       LSS_SC_BODY(3, ssize_t, 16, s, msg, flags);
     }
-    // TODO(csilvers): why is this ifdef'ed out?
-#if 0
     LSS_INLINE ssize_t LSS_NAME(sendto)(int s, const void *buf, size_t len,
                                         int flags,
                                         const struct kernel_sockaddr *to,
                                         unsigned int tolen) {
-      LSS_BODY(6, ssize_t, 11, s, buf, len, flags, to, tolen);
+      LSS_SC_BODY(6, ssize_t, 11, s, buf, len, flags, to, tolen);
     }
-#endif
     LSS_INLINE int LSS_NAME(shutdown)(int s, int how) {
       LSS_SC_BODY(2, int, 13, s, how);
